### PR TITLE
docs: adding an index for upgrade pages.

### PR DESCRIPTION
**Pull request by zbwright**
# Contributing guidelines

Please read our [contribution workflow][contributing] before submitting a pull request.

[contributing]: https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#contribution-flow

### CODE REVIEW

**Comment by heyitsanthony** [comment]
The `describes...` parts after the links seem redundant since the link already describes the target document.

**Comment by zbwright** [comment]
Agreed. We wanted it to be more than a list, but I didn't come up with anything more substantial. Maybe, for this page, it just remains a list? @joshix

File: Documentation/upgrades/upgrading-etcd.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Upgrading etcd clusters and applications
+
+This section contains documents specific to upgrading etcd clusters and applications.
+
+Moving from etcd API v2 to API v3
```
Comment (athai) [inline Documentation/upgrades/upgrading-etcd.md]: I think this may be easier to read if you bolded each subheading.

Comment (zbwright) [inline Documentation/upgrades/upgrading-etcd.md]: Also agreed. We don't have any of these sorts of pages yet. Part of this exercise will be defining the format.

Comment (joshix) [inline Documentation/upgrades/upgrading-etcd.md]: headings suffice

**Comment by heyitsanthony** [comment]
This should probably be linked in `docs.md` or have a `README.md` symlink pointing to itself in the `upgrades` directory since right now it's unreachable from any other repo doc

**Comment by zbwright** [comment]
The files linked from this page are all listed in docs.md, down at the bottom of the page under "Upgrading and compatibility". Should I add a link to this page as well?

**Comment by heyitsanthony** [comment]
@zbwright probably include it as an 'Upgrading' link under 'Maintenance' so the doc isn't dangling and remove the old upgrading section so there's no duplication?

**Comment by heyitsanthony** [comment]
@zbwright please squash and this should be good to merge. Thanks!

**Pull request merged**